generator/generator.go

```diff
@@ -38,7 +39,7 @@ type CaddyfileGenerator struct {
 
 // CreateGenerator creates a new generator
 func CreateGenerator(dockerClients []docker.Client, dockerUtils docker.Utils, options *config.Options) *CaddyfileGenerator {
-	var labelRegexString = fmt.Sprintf("^%s(_\\d+)?(\\.|$)", options.LabelPrefix)
+	var labelRegexString = fmt.Sprintf("^%s(_\\d+)?(\\.|$)", regexp.QuoteMeta(options.LabelPrefix))
```

Comment:
Here uses `regexp.QuoteMeta` to escapes all regular expression metacharacters in the user-provided label prefix, to make it a literal text in the `labelRegex`.